api/v1alpha1/addon_types.go

```diff
@@ -96,8 +102,6 @@ type GenericAddons struct {
 	ServiceLoadBalancer *ServiceLoadBalancer `json:"serviceLoadBalancer,omitempty"`
 }
 
-// +kubebuilder:validation:Optional
-// +kubebuilder:validation:Enum=ClusterResourceSet;HelmAddon
 type AddonStrategy string
```

Comment:
Moved this to the structs that are using `AddonStrategy` to be able to enforce the allowed strategies to only be  `HelmAddon`.
I first tried to "override" it just in `GenericCOSI` but both were still allowed.